src/main/java/com/jagrosh/jmusicbot/commands/music/SeekCmd.java

```diff
@@ -0,0 +1,74 @@
+package com.jagrosh.jmusicbot.commands.music;
+
+import com.jagrosh.jdautilities.command.CommandEvent;
+import com.jagrosh.jmusicbot.Bot;
+import com.jagrosh.jmusicbot.audio.AudioHandler;
+import com.jagrosh.jmusicbot.commands.MusicCommand;
+import com.jagrosh.jmusicbot.settings.Settings;
+import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
+import net.dv8tion.jda.core.Permission;
+
+import java.util.regex.Pattern;
+
+
+/**
+ * @author Whew., Inc.
+ */
+public class SeekCmd extends MusicCommand {
+    public SeekCmd(Bot bot) {
+        super(bot);
+        this.name = "seek";
+        this.help = "seeks the current song";
+        this.arguments = "<HH:MM:SS>|<MM:SS>|<SS>";
+        this.aliases = bot.getConfig().getAliases(this.name);
+        this.beListening = true;
+        this.bePlaying = true;
+    }
+
+    @Override
+    public void doCommand(CommandEvent event) {
+        AudioHandler handler = (AudioHandler) event.getGuild().getAudioManager().getSendingHandler();
+        if (handler.getPlayer().getPlayingTrack().isSeekable()) {
+            AudioTrack currentTrack = handler.getPlayer().getPlayingTrack();
+            Settings settings = event.getClient().getSettingsFor(event.getGuild());
+
+            if (!event.getMember().hasPermission(Permission.MANAGE_SERVER)
+                    && !event.getMember().getRoles().contains(settings.getRole(event.getGuild()))
+                    && currentTrack.getUserData(Long.class) != event.getAuthor().getIdLong()) {
+                event.replyError("You cannot seek **" + currentTrack.getInfo().title + "** because you didn't add it!");
+                return;
+            }
+
+            String args = event.getArgs();
+            long track_duration = handler.getPlayer().getPlayingTrack().getDuration();
+            int seek_milliseconds = 0;
+            int seconds;
+            int minutes = 0;
+            int hours = 0;
+
+            if (Pattern.matches("^([0-9]\\d):([0-5]\\d):([0-5]\\d)$", args)) {
+                hours = Integer.parseInt(args.substring(0, 2));
+                minutes = Integer.parseInt(args.substring(3, 5));
+                seconds = Integer.parseInt(args.substring(6));
+            } else if (Pattern.matches("^([0-5]\\d):([0-5]\\d)$", args)) {
+                minutes = Integer.parseInt(args.substring(0, 2));
+                seconds = Integer.parseInt(args.substring(3, 5));
+            } else if (Pattern.matches("^([0-5]\\d)$", args)) {
+                seconds = Integer.parseInt(args.substring(0, 2));
+            } else {
+                event.replyError("Invalid seek!");
+                return;
+            }
```

Comment:
I'm not a huge fan of having this here; instead maybe it could go in a `private int parseTime(String)` method that returns a number of seconds.

Alternatively, move the logic here as well as the time-formatting methods in FormatUtil to a new TimeUtil to make it easier to see the consistency of input and output.

Reply:
I went with the second option for now. It required some refactoring. Tell me what you think of it.